programmers-coding-test/wonji/greedy/1이될때까지.js

```diff
@@ -0,0 +1,18 @@
+const N = prompt(`나뉠 수 입력`);
+const K = prompt(`나눌 수 입력`);
+
+function solution(N,K){
+    let result = 0;
+    while (N > 1) {
+        if (N % K === 0 ){
+            N /= K;
+            result += 1
+        } else {
+            N -= 1
+            result += 1
+        }
+    }
+    return result;
```

Comment:
result 보단 count가 적절하다.
result += 1 은 중복된다. 어느 조건이든 수행하는 증감식으므로 조건 밖으로 빼주는 것이 적절하다.
if else가 단순하다면 삼항 연산자를 떠올려라. 가독성이 높아진다.

```suggestion
  let cnt = 0;
  while (N !== 1) {
    N = N % K ? N - 1 : N / K;  // N이 K로 나눠지지 않으면 N에서 1을 빼고, 나눠지면 N / K를 수행하여 N에 대입한다.
    cnt++;
  }
  return cnt;
```